android/app/src/main/java/com/mattermost/helpers/PushNotificationDataHelper.kt

```diff
@@ -33,83 +36,73 @@ class PushNotificationDataRunnable {
-                                val res = fetchTeamIfNeeded(db, serverUrl, teamId)
-                                teamData = res.first
-                                myTeamData = res.second
-                            }
+                        Log.i("ReactNative", "Start fetching notification data in server=$serverUrl for channel=$channelId")
 
-                            if (channelId != null) {
-                                val channelRes = fetchMyChannel(db, serverUrl, channelId, isCRTEnabled)
-                                channelData = channelRes.first
-                                myChannelData = channelRes.second
-                                loadedProfiles = channelRes.third
+                        val receivingThreads = isCRTEnabled && !rootId.isNullOrEmpty()
+                        val writableMap = Arguments.createMap()
 
-                                postData = fetchPosts(db, serverUrl, channelId, isCRTEnabled, rootId, loadedProfiles)
+                        if (!teamId.isNullOrEmpty()) {
+                            val res = fetchTeamIfNeeded(db, serverUrl, teamId)
+                            res.first?.let { writableMap.putMap("team", it) }
+                            res.second?.let { writableMap.putMap("myTeam", it) }
+                        }
 
-                                posts = postData?.getMap("posts")
-                                userIdsToLoad = postData?.getArray("userIdsToLoad")
-                                usernamesToLoad = postData?.getArray("usernamesToLoad")
-                                threads = postData?.getArray("threads")
-                                usersFromThreads = postData?.getArray("usersFromThreads")
+                        if (channelId != null && postId != null) {
+                            val channelRes = fetchMyChannel(db, serverUrl, channelId, isCRTEnabled)
+                            channelRes.first?.let { writableMap.putMap("channel", it) }
+                            channelRes.second?.let { writableMap.putMap("myChannel", it) }
+                            val loadedProfiles = channelRes.third
+
+                            // Fetch categories if needed
+                            if (!teamId.isNullOrEmpty() && writableMap.getMap("myTeam") != null) {
+                                // should load all categories
+                                val res = fetchMyTeamCategories(db, serverUrl, teamId)
+                                res?.let { writableMap.putMap("categories", it) }
+                            } else if (writableMap.getMap("channel") != null) {
+                                // check if the channel is in the category for the team
+                                val res = addToDefaultCategoryIfNeeded(db, writableMap.getMap("channel")!!)
+                                res?.let { writableMap.putArray("categoryChannels", it) }
+                            }
```

Comment:
Not sure, but I wonder if this will create any issue with DMs. The team id won't be there (I guess), so we should get the categories at least for the current team, to add it to the correct one, not the default one.

Reply:
in that case it adds the DM/GM to all teams